ibis/filesystems.py

```diff
@@ -278,42 +285,19 @@ def chown(self, path, owner=None, group=None):
 
     @implements(HDFS.exists)
     def exists(self, path):
-        try:
-            self.client.status(path)
-            return True
-        except Exception:
-            return False
+        return not self.client.status(path, strict=False) is None
 
     @implements(HDFS.ls)
     def ls(self, hdfs_path, status=False):
-        contents = self.client.list(hdfs_path)
-        if not status:
-            return [path for path, detail in contents]
-        else:
-            return contents
+        return self.client.list(hdfs_path, status=status)
 
     @implements(HDFS.mkdir)
     def mkdir(self, dir_path, create_parent=False):
-        # ugh, see #252
-
-        # create a temporary file, then delete it
-        dummy = pjoin(dir_path, util.guid())
-        self.client.write(dummy, '')
-        self.client.delete(dummy)
+        self.client.makedirs(dir_path) # TODO: What is `create_parent` for?
```

Comment:
Yes - intermediary directories get created as needed on HDFS (i.e. as if `-p` was always set). Would you like me to remove the `create_parent` argument or keep it as is (for backwards compatibility and possible future use)?